index.html

```diff
@@ -1799,39 +1812,40 @@ <h3>Securing Verifiable Credentials</h3>
 
         <p>
 At least one securing mechanism, and the details necessary to evaluate it, MUST
-be expressed for a <a>credential</a> or <a>presentation</a> to be a
-<a>verifiable credential</a> or <a>verifiable presentation</a>; that is, to be
-<a>verifiable</a>.
+be expressed for a <a>verifiable credential</a> or a
+<a>verifiable presentation</a>.
         </p>
         <p>
```

Comment:
```suggestion
      <p class="note"
        title="Being verifiable does not require successful verification">
"Verifiable credentials" and "verifiable presentations" MUST be securable, but MAY be unsecured. The point of this verifiability is the ability to test, via the verification process, whether the "verifiable credential" or "verifiable presentation" was secured, and, if so, whether its content has been tampered with. It does not matter whether this process succeeds or not, nor whether a proof or related mechanism is actually attached; it only matters whether the document is structured such that a proof or related mechanism _can be_ attached and used, as described by this specification.
        </p>
        <p>
```

Reply:
Hrm, I think this interpretation is what's tripping us up. IOW, this interpretation seems to be what @OR13 keeps objecting to. What I'm trying to do in this PR is make it clear that if you want to call something a VC/VP, it MUST be secured. 

What about something like:

```suggestion
      <p class="note"
        title="Being verifiable does not require successful verification">
"Verifiable credentials" and "verifiable presentations" MUST be secured. The point of this verifiability is the ability to test, via the verification process, whether the "verifiable credential" or "verifiable presentation" was secured, and, if so, whether its content has been tampered with. It does not matter whether this process succeeds or not; it only matters that the document was secured in some way where the verification process can be executed.
        </p>
        <p>
```

Reply:
"Verifiable credentials" and "verifiable presentations" MUST be secured. 
+1 to this. We have already said that this securing mechanism differentiates the credential from the verifiable credential.

Reply:
>> "Verifiable credentials" and "verifiable presentations" MUST be secured.

No. Very much, no.

> We have already said that this securing mechanism differentiates the credential from the verifiable credential.

Yes, the securing mechanism, which works out to the ability to attach a proof (internally or externally) and relevant details/data, which can then be used to test whether the payload (the collection of claims) is as issued or has been tampered with.

But that test does *not* need to succeed.

Real world example: US Birth and Death Certificates *may* be certified but this is not a requirement of all copies thereof. Many uses of these documents require a certified copy but many other uses do *not* require the certification. The difference between them typically amounts to the certified (a/k/a secured) a stamp (akin to if not actually a notary stamp), usually a signature, sometimes some special stationery. Photocopies of the certified have *exactly* the same information, but do not have an original signature, nor the stamp, nor the stationery.

Reply:
I think we're getting tripped up on what it means to be "secured"... I'm finding myself agreeing w/ both @David-Chadwick and @TallTed -- which might mean that this is a term/definition problem (which is true for many disagreements). :)

I think we agree? maybe? That in order for a "credential" to be a "verifiable credential", it MUST be secured using a securing mechanism. Simultaneously, we are NOT saying that verification of the securing mechanism has to succeed... we're just saying: "It has to be there in a way that a known verification process can be run against it."

I think some in the group are objecting to the notion that we can call something a "verifiable credential" when it isn't secured. So, those folks are drawing a distinction between "to be able to be secured" and "is secured, but might not verify" -- I think the group is pushing for the latter. I don't know if you, @TallTed, are pushing for the former?

In any case, just an observation that it's the "secured" language that might be tripping up consensus on this item?

Reply:
> I don't know if you, @TallTed, are pushing for ["to be able to be secured"]?

Yes, I am.

A door with a lock that is not locked is still a lockable door. Once locked, the lock might require a code on a keypad (internal) or a physical key/card (external) to be unlocked. It is typically obvious upon cursory inspection which of these is needed, and so the entity desiring entry knows what next-steps to take.

According to the VCDM, a verifiable {credential|presentation} must conform to a defined structure (Compacted JSON-LD), which must include two objects: a `@context` and a `@type`. Everything else is optional — including `proof`, because an external mechanism is allowed; an internal mechanism is not required.

Now, if you leave out `proof` because you intend to use an external mechanism, and then you fail — for any reason — in your use of that external mechanism, verification *will* fail — as intended and designed!  Verifiability is not lost; rather, this failure is confirmation that verifiability is working!

I *might* be convinced to require declaration of securing mechanism — at least when `proof` is not used for an internal mechanism, i.e., when an external mechanism (and proof material) is intended — as this would narrow the testing which a verifier needs to perform.

It seems to me that failure to attach/declare an external securing mechanism *and* failure to fill in `proof` serves as an obvious and quick verification failure.  But the other data in the VP/VC *might* still be considered *valid* in some scenarios.... This is up to the verifier and their business rules.

Reply:
> I might be convinced to require declaration of securing mechanism — at least when proof is not used for an internal mechanism, i.e., when an external mechanism (and proof material) is intended — as this would narrow the testing which a verifier needs to perform.

Can you elaborate on how you would see that working?

Reply:
From the current draft:
`At least one securing mechanism, and the details necessary to evaluate it, MUST
be expressed to construct a <a>verifiable credential</a> or a
<a>verifiable presentation</a>.`

It seems it's already expected to be working... for what is "expressing" a securing mechanism other than declaring it? `alg=none` might be the only detail the verifier needs to evaluate the mechanism according to *their* business logic.

But I stand by my assertion that, for instance, `ROT13` (or heck, `ROT0`) is a securing mechanism which can be expressed with no additional details necessary for its evaluation, and can be used to construct a VC or VP.

Reply:
Perhaps we can just make a statement like this: "A <a>verifiable credential</a> without a securing mechanism (neither internal nor external), MUST (will?) fail verification."?

Reply:
I could probably be convinced to something like that (reserving some language tweaks).

What does it do with `alg=none` (or whatever the closest parallel would be)? Is a mechanism with a weak (or `NULL`) key (or password) considered to be present or absent?

Reply:
> I could probably be convinced to something like that (reserving some language tweaks).
> 
> What does it do with `alg=none` (or whatever the closest parallel would be)? Is a mechanism with a weak (or `NULL`) key (or password) considered to be present or absent?

Absent. `alg=none` has no place in the VC Ecosystem.

Reply:
That would seem to cover a NULL key or password.

What about a weak key, password, mechanism, etc? How is this (particularly mechanism, where they may be broken after VC issuance and possibly after demise of VC issuer -- so reissue/update is not possible) to be monitored/maintained?

Reply:
> What about a weak key, password, mechanism, etc? How is this (particularly mechanism, where they may be broken after VC issuance and possibly after demise of VC issuer -- so reissue/update is not possible) to be monitored/maintained?

The topic above is important, and is probably something for the Security Considerations section. We kinda-sorta talk about it here, but clearly (given the above) the section could be improved:

https://w3c.github.io/vc-data-model/#cryptography-suites-and-libraries

... but "solving it" in this PR is deviating from the purpose of the PR, IMHO. Would you be ok w/ raising an issue on this topic? It requires discussion before we try to write anything into the spec.

Reply:
One of the purposes of this PR was to differentiate between a credential and a verifiableCredential, so agreeing on what a verifiableCredential is, is critical to this PR.

Reply:
I have reworked the language to specify that the thing that differentiates a "credential" from a "verifiable credential" is the fact that the latter is secured in a way that is cryptographically verifiable, and the former is not. Requesting re-reviews from @TallTed @David-Chadwick and @OR13.

Reply:
If your rework is lines 641-654 above, I'm OK with it.